crates/circuit/src/circuit_data.rs

```diff
@@ -1172,11 +1171,8 @@ impl CircuitData {
 
     /// Primary entry point for appending an instruction from Python space.
     pub fn append(&mut self, value: &Bound<CircuitInstruction>) -> PyResult<()> {
-        let py = value.py();
-        let new_index = self.data.len();
-        let packed = self.pack(py, &value.borrow())?;
-        self.data.push(packed);
-        self.track_instruction_parameters(new_index)
```

Comment:
Why do we no longer have to track the instruction parameters? If the block contains free parameters, don't we still have to track them?